### ISSUE TITLE
Add Set.Extra.symmetricDifference, isSuperset, areDisjoint

### ISSUE THREAD

**Issue by gampleman** [elmcraft/core-extra#38]
## `symmetricDifference : Set a -> Set a -> Set a`

The symmetric difference between two sets is a set that contains all the elements that are in one of the two sets, but not both.

(image of Venn diagram such as:

![](https://assets-eu-01.kc-usercontent.com/b94b6479-b913-0158-a9da-42d7ed61aab7/2be309dc-8a50-4ede-8ba9-f57addcf8d14/sql_joins_symmetric_difference.webp?w=2402&h=1256&auto=format&fit=crop)

could perhaps be useful).

## `isSuperset : Set a -> Set a -> Bool`

A set is a superset of another set if all the elements in the second set appear in the first set.

## `isSubset : Set a -> Set a -> Bool`

This is the existing `Set.Extra.subset`, but with a better name.

## `areDisjoint : Set a -> Set a -> Bool`

A set is disjoint from another set if they have no elements in common.

### Motivating use case 

These are fairly standard set operations that just seem missing. Symmetric difference is basically an SQL `FULL OUTER JOIN`.  `areDisjoint` can easily come up if the user is supposed to select checkboxes in two lists, but can't select an item in both lists.

The weakest is `isSuperset`, since it's exactly the same as `isSubset`, but with the arguments flipped.

### Rationale

Implementing these in user land is simple, albeit I'd bet more efficient algorithms exist:

```elm
symmetricDifference a b = 
   Set.difference (Set.union a b) (Set.intersection a b)

isSuperset a b =
   isSubset b a

areDisjoint a b =
   Set.intersection a b == Set.empty
```

Also, JavaScript is getting these [natively soon](https://www.sonarsource.com/blog/union-intersection-difference-javascript-sets/).

**Comment by gampleman** [comment]
@miniBill how does the `Of` suffix clarify the order? Is is still the case that `isSubsetOf a b --> True` means `a` is a subset of `b`?

I find the Of suffix more confusing in pipeline style:

```elm
a
    |> Set.isSubsetOf b
```

is more confusing than

```elm
a 
   |> Set.isSubset b
```

we could of course flip the arguments vs the existing function while we're making this change....

**Comment by miniBill** [comment]
I read `a |> Set.isSubsetOf b` as "a is a subset of b", which is what I think the implementation should do.

I see you've fixed it in #39
